src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -947,10 +947,22 @@ private bool IsAborted()
             return _disposed;
         }
 
+
+        // Check if lifetime expired on connection.
+        public bool LifetimeExpired(DateTimeOffset now, TimeSpan lifetime)
```

Comment:
Since we duplicate this function for both HttpConnection and Http2Connection, can we move it to HttpConnectionBase? And the CreationTime property too.

Reply:
There are other common properties. Should I move them too? I did look at HttpConnectionBase at the beginning but it looked like empty sub.

Reply:
My main concern is, don't duplicate the code.

We could solve this by moving this logic to HttpConnectionBase, or we could solve it by moving it somewhere else, like HttpConnectionPool. I'm happy either way.

HttpConnectionBase is pretty useless right now, and I'm not sure we really need it at all. So maybe it's better to not move the code there.

Reply:
I like you original proposal so I decided to go with it. Maybe we can make HttpConnectionBase more useful over time. It feels like there should be some commonalities.